common/src/main/java/com/microsoft/identity/common/internal/msafederation/FederatedCredential.kt

```diff
@@ -22,8 +22,17 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.msafederation
 
+import com.google.gson.annotations.SerializedName
+import java.util.AbstractMap
+
 /**
  * Represents credential artifact as result of successful sign in into a federated sign in provider
  * (Google/Apple). It can contain id token and/or auth code. See implementations for more details.
  */
-abstract class FederatedCredential(val federatedSignInProviderName: FederatedSignInProviderName)
+abstract class FederatedCredential(@SerializedName("signInProviderName") val signInProviderName: FederatedSignInProviderName) {
```

Comment:
```suggestion
abstract class MsaFederatedCredential(@SerializedName("signInProviderName") val signInProviderName: FederatedSignInProviderName) {
```

Reply:
Suggest rename to avoid any confusion about AAD which is where the word `federation` mostly gets used.

Reply:
renamed